fxgl/src/main/kotlin/com/almasb/fxgl/dsl/components/LiftComponent.kt

```diff
@@ -18,104 +18,95 @@ import javafx.util.Duration
+    private fun initTimer(liftData: LiftData) {
+        liftData.timer = FXGL.newLocalTimer()
+        liftData.timer.capture()
     }
 
     override fun onUpdate(tpf: Double) {
-        if (isHorizontal) {
-            if (timerX.elapsed(durationX)) {
-                isGoingRight = !isGoingRight
-                timerX.capture()
+        if (liftDataX.isHorizontal) {
+            if (liftDataX.timer.elapsed(liftDataX.duration)) {
+                liftDataX.isGoingPositive = !liftDataX.isGoingPositive
+                liftDataX.timer.capture()
             }
 
-            entity.translateX(if (isGoingRight) speedX * tpf else -speedX * tpf)
+            entity.translateX(if (liftDataX.isGoingPositive) liftDataX.speed * tpf else -liftDataX.speed * tpf)
         }
 
-        if (isVertical) {
-            if (timerY.elapsed(durationY)) {
-                isGoingUp = !isGoingUp
-                timerY.capture()
+        if (!liftDataY.isHorizontal) {
+            if (liftDataY.timer.elapsed(liftDataY.duration)) {
+                liftDataY.isGoingPositive = !liftDataY.isGoingPositive
+                liftDataY.timer.capture()
             }
 
-            entity.translateY(if (isGoingUp) -speedY * tpf else speedY * tpf)
+            entity.translateY(if (liftDataY.isGoingPositive) -liftDataY.speed * tpf else liftDataY.speed * tpf)
         }
     }
 
     fun xAxisDistanceDuration(distance: Double, duration: Duration) = this.apply {
-        distanceX = distance
-        durationX = duration
-        speedX = distance / duration.toSeconds()
+        liftDataX.distance = distance
+        liftDataX.duration = duration
+        liftDataX.speed = distance / duration.toSeconds()
 
-        isHorizontal = true
+        liftDataX.isHorizontal = true
     }
 
     fun xAxisSpeedDuration(speed: Double, duration: Duration) = this.apply {
-        speedX = speed
-        durationX = duration
-        distanceX = speed * duration.toSeconds()
+        liftDataX.speed = speed
+        liftDataX.duration = duration
+        liftDataX.distance = speed * duration.toSeconds()
 
-        isHorizontal = true
+        liftDataX.isHorizontal = true
     }
 
     fun xAxisSpeedDistance(speed: Double, distance: Double) = this.apply {
-        speedX = speed
-        distanceX = distance
-        durationX = Duration.seconds(distance / speed)
+        liftDataX.speed = speed
+        liftDataX.distance = distance
+        liftDataX.duration = Duration.seconds(distance / speed)
 
-        isHorizontal = true
+        liftDataX.isHorizontal = true
     }
 
     fun yAxisDistanceDuration(distance: Double, duration: Duration) = this.apply {
-        distanceY = distance
-        durationY = duration
-        speedY = distance / duration.toSeconds()
+        liftDataY.distance = distance
+        liftDataY.duration = duration
+        liftDataY.speed = distance / duration.toSeconds()
 
-        isVertical = true
+        liftDataY.isHorizontal = false
```

Comment:
Please see earlier comment. This, I  think, should be `liftDataY.isOn = true`

Reply:
I agree.